### PR TITLE
Update dependencies handling for the demo run

### PR DESCRIPTION
**Pull request by danielmellado**
This commit cleans up a all the dependency handling and automates
it. Besides, it also cleans the docs a bit and clarifies how to
run this.

Besides that, it also adds a github workflow for linting.

Signed-off-by: Daniel Mellado <dmellado@redhat.com>

### CODE REVIEW

**Comment by maiqueb** [comment]
I'd rather not have random binaries shipped via this repo. 

I don't mind providing the same via pulling the src code (gitignoring it), and building the binaries from it. 

... I know, few extra steps, but, I wouldn't just run a random binary I got from a github repo. Especially when it requires root access ...

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+      - name: install tox
+        run: python -m pip install --upgrade tox virtualenv setuptools pip
+      - name: run linter
+        run: tox -ebashate
```
Comment (maiqueb) [inline .github/workflows/lint.yml:22]: shouldn't we run the examples ?

File: README.md [hunk 2]
```diff
@@ -1,15 +1,18 @@
 # fosdem2023-cni-unchained
+
```
Comment (maiqueb) [inline README.md:2]: all changes in this file are not "seen" by the render - i.e. the same page is rendered with / without these changes. 

I don't think they're really required.

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+         -not \( -type d -name .?\* -prune \)    \
+         -type f                                 \
+         -not -name \*~                          \
+         -not -name \*.md                        \
+         \(                                      \
+          -name \*.sh                            \
+         \)                                      \
+         -print0 | xargs -0 bashate -v -iE006 -eE005,E042"
+
+[testenv:run_unlimited_bw_demo]
```
Comment (maiqueb) [inline tox.ini:26]: shouldn't we have a target for the other bandwidth scenario ?

**Review by maiqueb — CHANGES_REQUESTED** [review]
Huge improvement on the install dependency front.

Just a couple of suggestions / questions.

Thanks !

File: README.md (hunk 2 above)
Comment (danielmellado) [reply README.md:2]: Just for the sake of markdownlint and my ease of mind xD

File: .github/workflows/lint.yml (hunk 1 above)
Comment (danielmellado) [reply .github/workflows/lint.yml:22]: not really a need, as I don't think we should run the examples in a PR basis

File: tox.ini (hunk 3 above)
Comment (danielmellado) [reply tox.ini:26]: I'll hold onto that for when we're done with all the examples. I'll tackle whatever more needed ones in a follow-up PR.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (maiqueb) [reply .github/workflows/lint.yml:22]: Why wouldn't we ? If we have something that assures a PR does not break the examples, shouldn't we rely on it ?

**Review by maiqueb — APPROVED** [review]
OK, given https://github.com/maiqueb/fosdem2023-cni-unchained/pull/2#discussion_r1082700136, let's get this in, and add the tests / other config types afterwards.

**Pull request merged**